engine/src/main/java/org/pentaho/di/trans/steps/formula/RowForumulaContext.java

```diff
@@ -57,6 +61,16 @@ public RowForumulaContext( RowMetaInterface row ) {
   }
 
   public Type resolveReferenceType( Object name ) {
+    if ( name instanceof String ) {
+      ValueMetaInterface valueMeta = this.rowMeta.searchValueMeta( (String) name );
+      if ( valueMeta != null ) {
```

Comment:
If we are just validating for type String, can this be:
if(valueMeta != null && valueMeta.getType.equals(ValueMetaInterface.TYPE_STRING)) ?
this way we would have a simple if statement and not the switch.

Reply:
I wrote with a switch case because this can growth with different types. I think using a switch statement can be more useful on future.

Reply:
I check this cause a sonnar issue, i am going to convert this into a if statement